### ISSUE TITLE
Gradient checkpointing: Wrap `tf.keras.Model` or `tf.keras.layers.Layer` in `tf.recompute_gradients()`

### ISSUE THREAD

**Issue by jarednielsen** [tensorflow/tensorflow#36968]
I'm implementing gradient checkpointing with my Tensorflow 2.1 project, following the doc here: https://www.tensorflow.org/api_docs/python/tf/recompute_grad

When I try
```code
model = tf.recompute_grad(model)
```
it fails, and likewise with 
```code
model.layer = tf.recompute_grad(model.layer)
```

I see that Keras acts differently because tf.recompute_grad() wraps the function with an inner() call, but what should I do to get it working?

**Comment by mathemakitten** [comment]
+1 to this question, I'm doing something similar with `tf-nightly==2.2.0.dev20200119` with custom layers inherited from Keras layers and the docs could use some clarification on "keep a reference to the underlying object around for the purpose of accessing these variables"

**Comment by nickfrosst** [comment]
+1 to this. would be great to know how to make use of both keras layers and the gradient checkpointing functionality.

**Comment by nyngwang** [comment]
no reply.

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
